### PR TITLE
Corporate names in brackets are valid corporate authors

### LINKED ISSUE #3928: Integrity check should not report Coorporate Authors as "Names are not in standard format"
Jabref Master:

1. Add a corporate author in braces in the author field  e.g. `{JabRef}`
=> Integrity check reports: Names are not in standard biblatex format
Expected behaviour: No warning, as this is the correct way for corporate authors:
See section 2.3.3 Corporate Authors and Editors of the [biblatex manual](ftp://ftp.rrzn.uni-hannover.de/pub/mirror/tex-archive/macros/latex/contrib/biblatex/doc/biblatex.pdf)

Addtionally, JabRef should offer the field `shortauthor`

### PR DESCRIPTION
**Pull request by stefan-kolb**
Resolves #3928

### CODE REVIEW

**Comment by stefan-kolb** [comment]
AuthorList behaves correctly here. This is a problem of the PersonNamesChecker.

**Comment by stefan-kolb** [comment]
@tobiasdiez I changed the implementation now and added additonal test.

File: src/test/java/org/jabref/logic/integrity/PersonNamesCheckerTest.java [hunk 1]
```diff
@@ -31,4 +51,15 @@ public void complainAboutPersonStringWithTwoManyCommas() throws Exception {
...
     }
+
+    @Test
+    public void validCorporateNameInBrackets() throws Exception {
+        assertEquals(Optional.empty(), checker.checkValue("{JabRef}"));
+    }
+
+    @Test
+    public void validCorporateNameAndPerson() throws Exception {
+        assertEquals(Optional.empty(), checker.checkValue("{JabRef} and Stefan Kolb"));
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/integrity/PersonNamesCheckerTest.java]: Corporate authors at the end and in the middle work then also, right?

Comment (stefan-kolb) [reply src/test/java/org/jabref/logic/integrity/PersonNamesCheckerTest.java]: I guess so. It only depends on the AuthorsList Parser and the BracketRemover. I just added a few tests here to double test the scenario.

File: src/test/java/org/jabref/model/entry/AuthorListTest.java [hunk 2]
```diff
@@ -375,6 +374,16 @@ public void testCompanyAuthor() {
...
     }
 
+    @Test
+    public void testCompanyAuthorAndPerson() {
+        Author author = AuthorList.parse("{JabRef Developers} and Stefan Kolb").getAuthor(0);
+        Author expected = new Author(null, null, null, "JabRef Developers", null);
+        assertEquals(expected, author);
+        author = AuthorList.parse("{JabRef Developers} and Stefan Kolb").getAuthor(1);
+        expected = new Author("Stefan", "S.", null, "Kolb", null);
+        assertEquals(expected, author);
```
Comment (tobiasdiez) [inline src/test/java/org/jabref/model/entry/AuthorListTest.java]: Why not simply
`assertEquals(Arrays.asList(new Author(JabRef), new Author(Kolb)), AuthorList.parse(".."))`?

**Pull request merged**